RadixWallet/Features/ImportMnemonic/VerifyMnemonic/VerifyMnemonic.swift

```diff
@@ -0,0 +1,93 @@
+import ComposableArchitecture
+import SwiftUI
+
+// MARK: - DisplayMnemonic
+public struct VerifyMnemonic: Sendable, FeatureReducer {
+	public struct State: Sendable, Hashable {
+		public static let numberOfWordsToConfirm = 3
+
+		public let mnemonic: Mnemonic
+		public let wordsToConfirm: NonEmpty<IdentifiedArrayOf<OffsetIdentified<Mnemonic.Word>>>
+		public var enteredWords: IdentifiedArrayOf<OffsetIdentified<String>>
+		public var focusedField: Int?
+		public var invalidMnemonic: Bool = false
+
+		public init(mnemonic: Mnemonic) {
+			self.mnemonic = mnemonic
+
+			let identifiedWords = mnemonic.words.identifiablyEnumerated()
+			let checksumWord = identifiedWords.last!
+			let randomWords = identifiedWords
+				.dropLast() // without checksum word
+				.shuffled() // randomize
+				.prefix(Self.numberOfWordsToConfirm) // take the required number of words
+				.sorted { $0.offset < $1.offset } // sort after shuffling
+
+			self.wordsToConfirm = .init(randomWords + [checksumWord])!
+			self.enteredWords = wordsToConfirm.rawValue.map {
+				OffsetIdentified(offset: $0.offset, element: "")
+			}.asIdentifiable()
+			self.focusedField = wordsToConfirm.first?.offset
+		}
+	}
+
+	public enum ViewAction: Sendable, Equatable {
+		case wordChanged(OffsetIdentified<String>)
+		case wordSubmitted
+		case textFieldFocused(Int?)
+		case confirmSeedPhraseButtonTapped
+	}
+
+	public enum DelegateAction: Sendable, Equatable {
+		case mnemonicVerified
+	}
+
+	@Dependency(\.overlayWindowClient) var overlayWindowClient
+
+	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case let .wordChanged(identifiedWord):
+			// reset the invalidMnemonic state
+			state.invalidMnemonic = false
+			state.enteredWords.updateOrAppend(identifiedWord)
+			return .none
+		case let .textFieldFocused(focus):
+			// Don't set focus to nil if it will be changed to another field
+			if focus == nil {
```

Comment:
This is easier to read IMO:

```swift
let done = state.enteredWords.reduce(true) { $0 && !$1.element.isEmpty }

if let focus {
	state.focusedField = focus
} else if done {
	state.focusedField = nil
}
return .none
```

Reply:
`done` is less misleading than `didEnterAllWords`, because `didEnterAllWords` make it sound like all inputs are known to be valid BIP39 words, but no such validation has (yet) happened right.

Reply:
`element.isEmpty` appears 3 times in code, might be worth adding an extension:

```swift
extension OffsetIdentified where Element: Collection {
    var isEmpty: Bool { element.isEmpty }
}
```